### ISSUE TITLE
Upgrade NetData agent to version 1.21 in all of the demo sites.

### ISSUE THREAD

**Issue by knatsakis** [netdata/netdata#7848]
##### Feature idea summary
We need to upgrade the NetData agent on all of our demosites to 1.21 with eBPF enabled.

Also, please consider adding the upgrade step to our 'Releasing the Agent' documentation (so that we are reminded to upgrade after each release) and/or automating the upgrade process.

**WARNING:** *This excludes the London demo site, which we need to upgrade _manually and not automatically_*

For each demo site there will be a snapshot taken of the current state and it will go offline for a few minutes.

##### Expected behavior
All demo sites are running the latest stable version of the agent.

**Comment by prologic** [comment]
Couple of refinement points here:

- Can we enumerate the list of all demo sites/machines we currently have?
- Are these managed by code today? If not let's take this opportunity to Terraform it?
  - Same as above -- We can (_if not already_) just add this to our CI/CD pipelines and fully automate it.

**Comment by Ferroin** [comment]
Enumerating all of them is easy, all of them are listed on the main page of https://www.netdata.cloud/.

**Comment by prologic** [comment]
Well I have access to the machines now Thanks @knatsakis -- We can at least enuemrate what they're all running now :)

```sh
$ orgalorg -q -s -C /usr/sbin/netdata -v < ...
netdata v1.19.0-36-ga2b0c9c
netdata v1.16.1-1-g2db6d75
netdata v1.19.0-36-ga2b0c9c
netdata v1.16.0-15-gef94259
netdata v1.19.0-43-g14b1207
netdata v1.17.1-70-g9094de0f
netdata v1.16.0-15-gef94259a
netdata v1.19.0-43-g14b1207
netdata v1.17.1-70-g9094de0
netdata v1.16.1-1-g2db6d75
```

**Comment by prologic** [comment]
Unless there's a special reason we need to run different versions for some reason or another (_eg development/testing purposes_) I'd just Terraform this and stick it into out CI/CD flows so they're always kept up-t-date. Developers are always free to spin up their own temporary machines for whatever reason and if we need to make this simpler we will :)

**Comment by prologic** [comment]
In addition; we should also properly manage DNS entries for these machines as well as the CloudFlare resources in front of them. Right now we have to do manual SSH configuration individually to get to the right machines.

**Comment by mfundul** [comment]
@prologic don't forget the LXC containers that run the netdata slaves, which are version 1.12 or something.

**Comment by prologic** [comment]
> @prologic don't forget the LXC containers that run the netdata slaves, which are version 1.12 or something.

Do not know about this (_thanks!_) -- Can you help me understand; Why are we doing this? Why LXC? What are these other LXC-managed instances used for? Thanks 😀

**Comment by mfundul** [comment]
> 
> 
> > @prologic don't forget the LXC containers that run the netdata slaves, which are version 1.12 or something.
> 
> Do not know about this (_thanks!_) -- Can you help me understand; Why are we doing this? Why LXC? What are these other LXC-managed instances used for? Thanks 😀

I know is that there are some scripts in the netdata-demo-site that set-up a server automatically. These scripts set-up LXC containers to create a streaming scenario with netdata master and slaves. I also know that @cakrit wants all netdata versions updated, that is both the master netdata binaries and the slave netdata binaries.

In my opinion those scripts need to be rewritten to maybe use docker and pull different kind of images, and install the agent via more streamlined ways. But that's probably not a priority I guess.

**Comment by prologic** [comment]
> In my opinion those scripts need to be rewritten to maybe use docker and pull different kind of images, and install the agent via more streamlined ways. But that's probably not a priority I guess.

It _might_ actually be valuable to do this; so yes maybe.

**Comment by prologic** [comment]
- [ ] make sure no site has memory mode = save or map explicitly set
- [ ] check /etc/netdata/stream.conf for the slaves
- [ ] ensure auto-updates on all demos (_except london_)

**Comment by prologic** [comment]
I've upgraded all I can and ensured auto-updates are working on all the demo sites we care about. I cannot upgrade octopus or stackdriver instances as I do not have physical access to those machines or backing VMs and risk ruining them. Closing this as "done" for now but eventually I'd like to redo our Demo sites entirely as well as "demo setups" of varying things to "show off" -- But build better/proper tooling around this so no-one has to go figure this out by hand next time.
